forward_engineering/helpers/updateHelper.js

```diff
@@ -402,72 +413,61 @@ const generateScript = (child, udtMap, data, column, mode) => {
-	}
+	const addedEntities = child?.properties?.entities?.properties?.added;
+	const modifiedEntities = child?.properties?.entities?.properties?.modified;
+	const deletedEntities = child?.properties?.entities?.properties?.deleted;
+	
+	const addedEntitiesScripts = handleChange(addedEntities, udtMap, getAdd, data)
+	const modifiedEntitiesScripts = handleChange(modifiedEntities, udtMap, getUpdate, data);
+	const modifiedEntitiesScript = handleChange(deletedEntities, udtMap, getDelete, data);
 
-	if (objectContainsProp(child, 'added')) {
-		alterScript = mergeArrays(alterScript, handleChange(child.added, udtMap, getAdd, data));
-	}
+    const addedContainers = child?.properties?.containers?.properties?.added;
+    const modifiedContainers = child?.properties?.containers?.properties?.modified;
+    const deletedContainers = child?.properties?.containers?.properties?.deleted;
 
-	if (objectContainsProp(child, 'deleted')) {
-		alterScript = mergeArrays(alterScript, handleChange(child.deleted, udtMap, getDelete, data));
+	let addedContainersScripts = [];
+	let modifiedContainersScripts = [];
+	let deletedContainersScript = [];
+	
+	if (!data?.scriptOptions?.containers?.skipModified) {
+		addedContainersScripts = generateScript(addedContainers?.items, udtMap, data, 'containers', 'add');
+		modifiedContainersScripts = generateScript(modifiedContainers?.items, udtMap, data, 'containers', 'update');
+		deletedContainersScript = generateScript(deletedContainers?.items, udtMap, data, 'containers', 'delete');
 	}
-
-	return alterScript;
+	
+    const addedViews = child?.properties?.views?.properties?.added;
+    const modifiedViews = child?.properties?.views?.properties?.modified;
+    const deletedViews = child?.properties?.views?.properties?.deleted;
+		
+    const addedViewsScripts = generateScript(addedViews?.items, udtMap, data, 'views', 'add');
+    const modifiedViewsScripts = generateScript(modifiedViews?.items, udtMap, data, 'views', 'update');
+    const deletedViewsScript = generateScript(deletedViews?.items, udtMap, data, 'views', 'delete');
+
+	const modelDefinitions = child?.properties?.modelDefinitions;
+    const sortAddedUdtResult = sortAddedUdt(modelDefinitions);
+    const addedModelDefinitions = sortAddedUdtResult?.properties?.added;
+    const modifiedModelDefinitions = sortAddedUdtResult?.properties?.modified;
+    const deletedModelDefinitions = sortAddedUdtResult?.properties?.deleted;
+	
+    const addedModelDefinitionsScripts = generateScript(addedModelDefinitions.items, udtMap, data, 'udt', 'add');
+    const modifiedModelDefinitionsScripts = generateScript(modifiedModelDefinitions.items, udtMap, data, 'udt', 'update');
+    const deletedModelDefinitionsScript = generateScript(deletedModelDefinitions.items, udtMap, data, 'udt', 'delete');
+	
+	return [
+			...modifiedEntitiesScripts,
+			...addedEntitiesScripts,
+			...modifiedEntitiesScript,
```

Comment:
Create separate functions that handle generation of scripts related to entities, containers, views etc

Reply:
done with this